R/PackageInheritanceReporter.R

```
@@ -0,0 +1,212 @@
+                    , lengthChange = FALSE
+                )
+            )
+            # Round the double columns to three digits for formatting reasons
+            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+            tableObj <- DT::formatRound(columns = numCols, table = tableObj
+                                        , digits=3)
+            return(tableObj)
+        }
+    ),
+
+    active = list(
+        nodes = function(){
+            if (is.null(private$cache$nodes)){
+                log_info("Extracting classes as nodes...")
+
+                pkg_env <- private$get_pkg_env()
+
+                nodeList <- list(data.table::data.table(
+                    node = character(0), classType = character(0)
+                ))
+
+                for (thisObjName in names(pkg_env)) {
+
+                    thisObj <- get(thisObjName, pkg_env)
+
+                    # S4 classes and References classes
+                    # Specified class name is the important name
+                    if (grepl("^\\.__C__", thisObjName) & isS4(thisObj)) {
+
+                        if (methods::is(thisObj, "refClassRepresentation")) {
+                            thisObjClassType <- "Reference"
+                        } else {
+                            thisObjClassType <- "S4"
+                        }
+
+                        nodeList <- c(nodeList, list(data.table::data.table(
+                            node = thisObj@className
+                            , classType = thisObjClassType
+                        )))
+
+                    # R6 classes
+                    # Generator object name is the important name
+                    } else if (R6::is.R6Class(thisObj)) {
+
+                        nodeList <- c(nodeList, list(data.table::data.table(
+                            node = thisObjName
+                            , classType = "R6"
+                        )))
+
+                    }
+
+                }
```

if all three checks failed, this would silently ignore all of this and probably break with some cryptic error further down. I think you should isolate the "figure out what type it is" all in one block at the top, and then do

```
nodeDT <- switch(
    thisObjClassType,
   "R6" = {stuff()},
   "S4" = {stuff()},
   "Reference" = {stuff()},
    NULL
)
```

---

After discussing this further with @jameslamb, this particular implementation isn't brittle but we still have some disagreements about general style of control flow.

I'm going to leave this implementation as-is because I don't feel like I know of a clean refactor that isn't more complicated or harder to read in a different way. I did add some comments to this method to overall make it more clear what it is doing.